lib/src/revset.rs

```diff
@@ -1817,6 +1817,22 @@ fn to_difference_range(
             heads: heads.clone(),
             generation: generation.clone(),
         })),
+        // ::heads & ~(::roots-) -> ::heads & ~ancestors(heads, 1..) -> roots-..heads
```

Comment:
nit: `ancestors(heads, 1..)` should be `ancestors(roots, 1..)`

Reply:
Oops, copy paste error. Fixed.